app/post.go

```diff
@@ -34,6 +34,25 @@ func CreatePostAsUser(post *model.Post) (*model.Post, *model.AppError) {
 		return nil, err
 	}
 
+	if utils.IsLicensed() && *utils.Cfg.TeamSettings.ExperimentalTownSquareIsReadOnly &&
```

Comment:
This is a bit redundant since it's immediately followed by a call to CreatePost which also checks this. It's also doing two database hits then to get the user twice. Can we instead check which error is returned by CreatePost below and use that to tell if we should make the ephemeral post?